src/platform_impl/macos/event_loop.rs

```diff
@@ -114,6 +164,49 @@ impl<T> EventLoop<T> {
     }
 }
 
+/// Catches panics that happen inside `f` and when a panic
+/// happens, stops the `sharedApplication`
+pub(crate) fn stop_app_on_panic<F: FnOnce() -> R + UnwindSafe, R>(f: F) -> Option<R> {
+    match catch_unwind(f) {
+        Ok(r) => Some(r),
+        Err(e) => {
+            // It's important that we set the panic before requesting a `stop`
+            // because some callback are still called during the `stop` message
+            // and we need to know in those callbacks if the application is currently
+            // panicking
+            {
+                let mut panic = CURRENT_PANIC.lock().unwrap();
+                if panic.is_none() {
+                    *panic = Some(e);
+                }
+            }
+            unsafe {
+                let app_class = class!(NSApplication);
+                let app: id = msg_send![app_class, sharedApplication];
+                let _: () = msg_send![app, stop: nil];
```

Comment:
I know this isn't consistently used yet, but this is equivalent:
```suggestion
                let () = msg_send![app, stop: nil];
```